boa/src/exec.rs

```diff
@@ -507,6 +507,39 @@ impl Executor for Interpreter {
                     ValueData::Function(_) => "function",
                 }))
             }
+            // <tc39.es/ecma262/#sec-try-statement>
+            ExprDef::TryCatch(ref try_block, ref catch_expr, ref finally_block) => {
+                if let Err(val) = self.run(try_block) {
+                    if let Some((catch_bind, catch_block)) = catch_expr {
+                        {
+                            let env = &mut self.realm.environment;
+                            env.push(new_declarative_environment(Some(
+                                env.get_current_environment_ref().clone(),
+                            )));
+                        }
+
+                        if let Some(catch_bind) = catch_bind {
+                            self.realm.environment.create_mutable_binding(
+                                catch_bind.clone(),
+                                false,
+                                VariableScope::Block,
+                            );
+
+                            self.realm.environment.initialize_binding(catch_bind, val);
+                        }
+
+                        todo!("run the catch block");
```

Comment:
Not sure if I'm doing this right, @jasonwilliams. The `catch_expr` here is a `(String, ExprDef::Block)` tuple, and I'm not sure how can I pass the binding value to the catch block. Am I doing this right? Should the type of expression be different?

Reply:
Seems like you've done it right to me, the catch_bind is the parameter name right?
That should be initiated on the new block.

You'll also need to pop that environment when you've finished with it but there's a lot of refactoring needed in that anyway

Reply:
About this, the catch block is a `Block` type expression (as it has been implemented right now) but I need to pass a variable to it. Is this the correct way to do it?